nbsmoke/__init__.py

```diff
@@ -140,7 +140,10 @@ def __init__(self, fspath, parent=None, config=None, session=None, dowhat=RunNb)
         super(IPyNbFile,self).__init__(fspath, parent=parent, config=None, session=None)
 
     def collect(self):
-        yield self._dowhat(str(self.fspath), self)
+        if hasattr(self._dowhat, "from_parent"):
+            yield self._dowhat.from_parent(self, fspath=self.fspath)
```

Comment:
Maybe

```suggestion
            yield self._dowhat.from_parent(self, name=str(self.fspath))
```

to address the error in travis:

```
==================================== ERRORS ====================================
______________________ ERROR collecting testing123.ipynb _______________________
/home/travis/build/pyviz-dev/nbsmoke/.tox/py37/lib/python3.7/site-packages/nbsmoke/__init__.py:144: in collect
    yield self._dowhat.from_parent(self, fspath=self.fspath)
/home/travis/build/pyviz-dev/nbsmoke/.tox/py37/lib/python3.7/site-packages/_pytest/nodes.py:195: in from_parent
    return cls._create(parent=parent, **kw)
/home/travis/build/pyviz-dev/nbsmoke/.tox/py37/lib/python3.7/site-packages/_pytest/nodes.py:99: in _create
    return super().__call__(*k, **kw)
E   TypeError: __init__() got an unexpected keyword argument 'fspath'
=========================== short test summary info ============================
ERROR testing123.ipynb - TypeError: __init__() got an unexpected keyword argu...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.29s ===============================
```

It's sure confusing :( `str(fspath)` is being used as `name`.

If anyone ever understood all of this at any point, they could address #29...although again, ultimately I hope to delete a lot of this project...

Reply:
This suggestion is now hidden/showing as "outdated" for me in github, but I think it's the right thing to do: see #44 where it worked (and where after updating for pytest land, I discover I also need to update for jupyter land...).